modal_test_support/functions.py

```diff
@@ -173,6 +173,42 @@ def generator(self, x):
         return self._generator(x)
 
 
+@stub.cls()
+class LifecycleCls:
+    """Ensures that {sync,async} lifecycle hooks work with {sync,async} functions."""
+
+    def __init__(self):
+        self.events = []
+
+    @enter()
+    def enter_sync(self):
+        self.events.append("enter_sync")
+
+    @enter()
+    async def enter_async(self):
+        self.events.append("enter_async")
+
+    # TODO: We have no way of verifying @exit() handler output in container_test.py right now.
```

Comment:
Should be possible when using `_run_container_process` (see the last few tests in container_test.py) since those use subprocesses for the container tests, although it would have to rely on printing stuff to stdout or similar to communicate since no more outputs are returned after exit handlers. 
We can also assert that the VolumeCommit rpc is called

Reply:
Thanks for the advice, I just added the new tests using `_run_container_process`!